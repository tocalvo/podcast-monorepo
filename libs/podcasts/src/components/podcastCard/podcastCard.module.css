.card {
  display: flex;
  flex: 1 1 175px;
  max-width: 175px;
  flex-direction: column;
  border-radius: 3%;
  align-items: center;
  margin-bottom: 80px;
  box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
  padding: 0px 10px;
  cursor: pointer;
}

.image {
  aspect-ratio: 1/1;
  border-radius: 50%;
  width: 100px;
  margin-top: -30px;
  /* to improve white background images contrast */
  border: 0.1px solid lightgrey;
}

.title {
  text-align: center;
  font-weight: 700;
}

.artist {
  text-align: center;
  color: dimgrey;
}
