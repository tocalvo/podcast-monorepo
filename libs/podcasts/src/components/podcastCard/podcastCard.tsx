import { ITunesPodcastEntry } from '../../models';
import styles from './podcastCard.module.css';
import { getImageFromEntry } from '../../utils';
import { Link } from 'react-router';

interface PodcastCardProps {
  podcast: ITunesPodcastEntry;
}
export const PodcastCard: React.FC<PodcastCardProps> = ({ podcast }) => {
  if (!podcast) {
    return; // return skeleton
  }
  const image = getImageFromEntry(podcast);
  return (
    <Link
      data-testid="podcast-card"
      className={styles['card']}
      aria-label="Ir al detalle del podcast"
      to={`/podcast/${podcast.id.attributes['im:id']}`}
    >
      <img className={styles['image']} alt="podcast logo" src={image.label} />
      <p className={styles['title']}>{podcast['im:name'].label}</p>
      <p className={styles['artist']}>Author: {podcast['im:artist'].label}</p>
    </Link>
  );
};

export default PodcastCard;
