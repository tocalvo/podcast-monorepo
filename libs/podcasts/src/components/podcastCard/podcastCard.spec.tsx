import { render, screen } from '@testing-library/react';

import { PodcastCard } from './podcastCard';
import { podcastMock } from '../../__mocks__/podcast.mock';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { BrowserRouter } from 'react-router';

describe('PodcastCard', () => {
  it('should render successfully', () => {
    const queryClient = new QueryClient();
    const { baseElement } = render(
      <BrowserRouter>
        <QueryClientProvider client={queryClient}>
          <PodcastCard podcast={podcastMock} />
        </QueryClientProvider>
      </BrowserRouter>
    );
    expect(baseElement).toBeTruthy();
  });

  it('should render the content', () => {
    const queryClient = new QueryClient();
    const { container } = render(
      <BrowserRouter>
        <QueryClientProvider client={queryClient}>
          <PodcastCard podcast={podcastMock} />
        </QueryClientProvider>
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });

  it('should have a link', () => {
    const queryClient = new QueryClient();
    render(
      <BrowserRouter>
        <QueryClientProvider client={queryClient}>
          <PodcastCard podcast={podcastMock} />
        </QueryClientProvider>
      </BrowserRouter>
    );
    const link = screen.getByRole('link', {
      name: 'Ir al detalle del podcast',
    });
    expect(link.getAttribute('href')).toBe('/podcast/1535809341');
  });
});
