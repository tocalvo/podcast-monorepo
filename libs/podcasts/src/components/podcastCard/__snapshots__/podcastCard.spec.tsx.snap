// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PodcastCard > should render the content 1`] = `
<div>
  <a
    aria-label="Ir al detalle del podcast"
    class="_card_aabbb0"
    data-discover="true"
    data-testid="podcast-card"
    href="/podcast/1535809341"
  >
    <img
      alt="podcast logo"
      class="_image_aabbb0"
      src="https://is1-ssl.mzstatic.com/image/thumb/Podcasts113/v4/f2/21/fa/f221fabd-017f-5125-633b-f1fe4f39802a/mza_182995249085044287.jpg/55x55bb.png"
    />
    <p
      class="_title_aabbb0"
    >
      The Joe Budden Podcast
    </p>
    <p
      class="_artist_aabbb0"
    >
      Author: 
      The Joe Budden Network
    </p>
  </a>
</div>
`;
